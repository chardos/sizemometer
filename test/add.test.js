const fs = require('fs-extra');
const { addConfigFile, writeFile } = require('./helpers');
const { cmd } = require('./wrappers');
const nodeCmd = require('node-cmd');
const { largeText, smallText } = require('./dummys/textFiles');
const { SIZES_JSON_PATH } = require('./constants');

describe('Command: Add', () => {
  // beforeEach(async () => {
  //   process.argv.splice(2);
  //   process.argv[2] = 'add'
  // });

  describe('when there is no existing history.json', () => {
    it('it should create a history.json with the first entry', async () => {
      await addConfigFile({files: ['dist/test.txt']});
      await writeFile({
        path: 'dist/test.txt',
        body: smallText
      });

      await cmd('sizemometer add').then((output) => {
        console.log('output:', output);
      });

      const jsonExists = await fs.exists(SIZES_JSON_PATH);
      expect(jsonExists).toEqual(true);

      const sizesJsonBuffer = await fs.readFile(SIZES_JSON_PATH);
      expect(
        sizesJsonBuffer.toString()
      ).toMatchSnapshot()
    })
  })

  describe('when there is a pre-existing history.json', () => {
    describe('and there is no new commits', () => {
    
    })  

    describe('and there is a new commit', () => {
    
    })  
  })

  describe('test multiple files', () => {
    
  })
})
