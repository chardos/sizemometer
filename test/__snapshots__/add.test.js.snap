// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command: Add when there is no existing history.json it should create a history.json with the first entry 1`] = `
"{
  \\"dist/test.txt\\": [
    {
      \\"author\\": \\"Richard Tan\\",
      \\"commitHash\\": \\"4999a423230ed65ed7eaf4218784606c28931e80\\",
      \\"commitMessage\\": \\"temp\\",
      \\"size\\": 80
    }
  ]
}"
`;
